### ISSUE TITLE
nvidia/canary-1b-v2 transcript is not normalized

### ISSUE THREAD

**Issue by kunibald413** [NVIDIA/NeMo#14566]
it is not fully clear how to have it return normalized transcript, could you assist?

running this code:

```python
from nemo.collections.asr.models import ASRModel

asr_ast_model = ASRModel.from_pretrained(model_name="nvidia/canary-1b-v2")
asr_ast_model.to("cuda")

output = asr_ast_model.transcribe([wav_path], source_lang='en', target_lang='en')
```

outputs:

```
output: [Hypothesis(score=0.0, y_sequence=tensor([ 3554,  2585,  2107,  3243, 16122, 16060, 10297,  1206, 12782, 10166,
         1392,  4551,  1612, 16122, 16060,  3024, 16053,  1124,  1123,  4187,
        16085,  1385, 16067, 16053,  1132,  4187, 16085,  1385, 16067, 16053,
         1127,  1130, 16067, 16053,  1130, 16073]), text="You can see there's big gaps between and then it's like 10 viewers, 9 viewers, 47, 7.", dec_out=None, dec_state=None, timestamp=[], alignments=None, frame_confidence=None, token_confidence=None, word_confidence=None, length=0, y=None, lm_state=None, lm_scores=None, ngram_lm_state=None, tokens=None, last_token=None, token_duration=None, last_frame=None)]
text: You can see there's big gaps between and then it's like 10 viewers, 9 viewers, 47, 7.
```

would expect 
text: `You can see there's big gaps between and then it's like ten viewers, nine viewers, fourty-seven, seven.`

same here:

```
output: [Hypothesis(score=0.0, y_sequence=tensor([16053,  1131,  1123,  1123, 16145, 16053,  1126,  1126, 16067,  1123,
         1123,  1123,  4187, 16085,  1385,  9003,  2472, 16073,  8936, 16067,
        10479,  1363, 16057, 16122, 16058,  2259,  5617,  1306,  1274,  1814,
         1731, 16145]), text="800? 33,000 viewers average. Well, why isn't this going on anymore?", dec_out=None, dec_state=None, timestamp=[], alignments=None, frame_confidence=None, token_confidence=None, word_confidence=None, length=0, y=None, lm_state=None, lm_scores=None, ngram_lm_state=None, tokens=None, last_token=None, token_duration=None, last_frame=None)]

text: 800? 33,000 viewers average. Well, why isn't this going on anymore?
```

would expect: `eight-hundred? thirty-three thousand viwers [...]`

-----

added print in `encode_turn`
```python
    def encode_turn(
        self, prompt_template: str, expected_slots: dict[str, Modality], slot_values: dict[str, Any]
    ) -> list[int]:
        prompt = prompt_template

        # normal case
        for slot in expected_slots:
            # For the final substitution of 'slot' in the template we have to mangle it to '|slot|' anyway,
            # but 'slot' form enables to use valid python identifiers as **kwargs
            # for passing slots around in user functions.
            value = slot_values.get(slot)
            assert value is not None, f"Missing required {slot=} in {slot_values=} for {prompt_template=}"
            prompt = prompt.replace(_mangled(slot), value)
        print(f"encode_turn prompt {prompt}") # added log
        return self._apply_tokenizer(prompt, lang=slot_values.get(self.PROMPT_LANGUAGE_SLOT))
```

prints out this:

```
encode_turn prompt <|startofcontext|><|startoftranscript|><|emo:undefined|><|en|><|en|><|pnc|><|noitn|><|notimestamp|><|nodiarize|>
```

hack chaging it to `<itn>` also does not seem to have any effect
```
encode_turn prompt <|startofcontext|><|startoftranscript|><|emo:undefined|><|en|><|en|><|pnc|><|itn|><|notimestamp|><|nodiarize|>
```

---

here are logs when loading the model
i additinoaly printed out the prompt and the default dialog

```python
asr_ast_model = ASRModel.from_pretrained(model_name="nvidia/canary-1b-v2")
asr_ast_model.to("cuda")

print(asr_ast_model.prompt)
print("get_default_dialog_slots: ", asr_ast_model.prompt.get_default_dialog_slots())
```

<details>
<summary>Click here to expand the logs!</summary>
[NeMo I 2025-08-24 15:36:43 nemo_logging:393] Tokenizer CanaryBPETokenizer initialized with 16384 tokens
[NeMo W 2025-08-24 15:36:43 nemo_logging:405] If you intend to do training or fine-tuning, please call the ModelPT.setup_training_data() method and provide a valid configuration file to setup the train data loader.
    Train config : 
    use_lhotse: true
    skip_missing_manifest_entries: true
    input_cfg: null
    tarred_audio_filepaths: null
    manifest_filepath: null
    sample_rate: 16000
    shuffle: true
    num_workers: 4
    pin_memory: true
    prompt_format: canary2
    max_duration: 40.0
    min_duration: 0.01
    text_field: answer
    lang_field: target_lang
    use_bucketing: true
    max_tps: null
    bucket_duration_bins: null
    bucket_batch_size: null
    num_buckets: null
    bucket_buffer_size: 20000
    shuffle_buffer_size: 10000
    
[NeMo W 2025-08-24 15:36:43 nemo_logging:405] If you intend to do validation, please call the ModelPT.setup_validation_data() or ModelPT.setup_multiple_validation_data() method and provide a valid configuration file to setup the validation data loader(s). 
    Validation config : 
    use_lhotse: true
    prompt_format: canary2
    manifest_filepath: null
    sample_rate: 16000
    batch_size: 4
    shuffle: true
    max_duration: 40.0
    min_duration: 0.1
    num_workers: 2
    pin_memory: true
    text_field: answer
    lang_field: target_lang
    
[NeMo I 2025-08-24 15:36:43 nemo_logging:393] PADDING: 0
[NeMo I 2025-08-24 15:36:55 nemo_logging:393] Tokenizer SentencePieceTokenizer initialized with 16384 tokens
[NeMo W 2025-08-24 15:36:58 nemo_logging:405] If you intend to do training or fine-tuning, please call the ModelPT.setup_training_data() method and provide a valid configuration file to setup the train data loader.
    Train config : 
    use_lhotse: true
    skip_missing_manifest_entries: true
    input_cfg: null
    tarred_audio_filepaths: null
    manifest_filepath: null
    sample_rate: 16000
    shuffle: true
    num_workers: 2
    pin_memory: true
    max_duration: 40.0
    min_duration: 0.1
    text_field: answer
    batch_duration: null
    max_tps: null
    use_bucketing: true
    bucket_duration_bins: null
    bucket_batch_size: null
    num_buckets: null
    bucket_buffer_size: 20000
    shuffle_buffer_size: 10000
    
[NeMo W 2025-08-24 15:36:58 nemo_logging:405] If you intend to do validation, please call the ModelPT.setup_validation_data() or ModelPT.setup_multiple_validation_data() method and provide a valid configuration file to setup the validation data loader(s). 
    Validation config : 
    use_lhotse: true
    manifest_filepath: null
    sample_rate: 16000
    batch_size: 16
    shuffle: false
    max_duration: 40.0
    min_duration: 0.1
    num_workers: 2
    pin_memory: true
    text_field: answer
    
[NeMo I 2025-08-24 15:36:58 nemo_logging:393] PADDING: 0
[NeMo I 2025-08-24 15:37:05 nemo_logging:393] Model EncDecCTCModelBPE was successfully restored from /root/.cache/huggingface/hub/models--nvidia--canary-1b-v2/snapshots/21c939b94d98894647bdc73fa86e19c458f7bfc9/canary-1b-v2.nemo.
[NeMo I 2025-08-24 15:37:08 nemo_logging:393] Model EncDecMultiTaskModel was successfully restored from /root/.cache/huggingface/hub/models--nvidia--canary-1b-v2/snapshots/21c939b94d98894647bdc73fa86e19c458f7bfc9/canary-1b-v2.nemo.
<nemo.collections.common.prompts.canary2.Canary2PromptFormatter object at 0x792223706ed0>
get_default_dialog_slots:  [{'role': 'user', 'slots': {'decodercontext': '', 'emotion': '<|emo:undefined|>', 'source_lang': '<|en|>', 'target_lang': '<|en|>', 'pnc': '<|pnc|>', 'itn': '<|noitn|>', 'timestamp': '<|notimestamp|>', 'diarize': '<|nodiarize|>'}}, {'role': 'user_partial', 'slots': {'decodercontext': ''}}]
</details>

**Comment by kunibald413** [comment]
@tbartley94 
any idea or any hint? `nvidia/canary-1b`  (v1 if i'm correct) transcribes normalized by default

**Comment by tbartley94** [comment]
@nithinraok @monica-sekoyan pretty sure y'all removed the PnC flag and just had PNC text over all data, correct?

**Comment by monica-sekoyan** [comment]
Hi @kunibald413 and @tbartley94 , yes, `Canary-1b-v2` is trained fully on PnC and ITN data, that is why you have transcripts with numbers (and other ITN aspects) and with Punctuation and Capitalization.

I believe for the future models we will enable switching between these states

**Comment by kunibald413** [comment]
so  `nvidia/canary-1b-v2` can't do text normalization is what i get from this

**Comment by nithinraok** [comment]
no option for normalization with canary-1b-v2, the output is based on training data.
